frontend/ui/widget/container/framecontainer.lua

```diff
@@ -90,6 +90,21 @@ function FrameContainer:onUnfocus()
     return false
 end
 
+local floor = math.floor
+local ceil = math.ceil
+
+local function paintRoundedRectRGB(bb, x, y, w, h, c, r)
+    x, y = ceil(x), ceil(y)
+    h, w = ceil(h), ceil(w)
+    if not r or r == 0 then
+        bb:paintRectRGB32(x, y, w, h, c)
+    else
+        if h < 2*r then r = floor(h/2) end
+        if w < 2*r then r = floor(w/2) end
+        bb:paintBorder(x, y, w, h, r, c, r)
+        bb:paintRectRGB32(x+r, y+r, w-2*r, h-2*r, c)
+    end
+end
```

Comment:
Yeah, this really needs to be in base.

Which will be mostly trivial (i.e., new RGB wrappers for paintRoundedRect & paintBorder), except for rounded rectangles, which currently doesn't have a color implementation.

Soooo, maybe needs a specific nerfed implementation of *paintBorder* that does *NOT* support a radius, instead of silzently producing garbage ;).

Reply:
As actual new implementations, I mean, don't try to finagle color handling in the existing methods ;).

(One, that'll be slightly nightmarish to deal with; two, I'd veto it because it's extra complexity for the sake of a super-mega-hyper-niche feature ;)).

i.e., code duplication there will be, I'm fine with it ;).